spring-cloud-kubernetes-fabric8-autoconfig/src/main/java/org/springframework/cloud/kubernetes/fabric8/Fabric8PodUtils.java

```diff
@@ -55,33 +55,25 @@ public Fabric8PodUtils(KubernetesClient client) {
 
 		this.client = client;
 		this.hostName = System.getenv(HOSTNAME);
-		this.current = LazilyInstantiate.using(() -> internalGetPod());
+		this.current = LazilyInstantiate.using(this::internalGetPod);
 	}
 
 	@Override
 	public Supplier<Pod> currentPod() {
 		return this.current;
 	}
 
-	@Override
-	public Boolean isInsideKubernetes() {
-		return currentPod().get() != null;
-	}
-
-	private synchronized Pod internalGetPod() {
+	private Pod internalGetPod() {
 		try {
-			if (isServiceAccountFound() && isHostNameEnvVarPresent()) {
+			if (isHostNameEnvVarPresent() && isServiceAccountFound()) {
```

Comment:
A simple `JMH` test shows that `System::getEnv` is 37 times faster then `Paths...`, so just let's revert their order. You can look here : https://github.com/wind57/0x-github-jmh-runner/pull/4/checks?check_run_id=1554720292 under `Run JMH tests`